src/python/pants/pantsd/service/pailgun_service.py

```diff
@@ -53,6 +54,7 @@ def _setup_pailgun(self):
     """Sets up a PailgunServer instance."""
     # Constructs and returns a runnable PantsRunner.
     def runner_factory(sock, arguments, environment):
+      start_time = time.time()
```

Comment:
Would be good to add a comment indicating how you chose this location for `start_time`: basically, it's a close to the connection arriving off the socket as we can get?

nit: Could use https://github.com/pantsbuild/pants/blob/6d87acdaec4b32155fa3caacf9a4c9a41adddb3f/src/python/pants/util/contextutil.py#L269 ? Maybe overkill, but.